django/contrib/postgres/signals.py

```diff
@@ -18,3 +19,20 @@ def register_hstore_handler(connection, **kwargs):
         # This is also needed in order to create the connection in order to
         # install the hstore extension.
         pass
+
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute("SELECT typarray FROM pg_type WHERE typname = 'citext'")
+            oids = tuple(row[0] for row in cursor)
+        array_type = psycopg2.extensions.new_array_type(oids, 'citext[]', psycopg2.STRING)
+        psycopg2.extensions.register_type(array_type, None)
+    except ProgrammingError:
+        # citext is not available on the database.
```

Comment:
Perhaps it would be enough to say, "citext is not available on the database. The same comments in the except block of the above call to register_hstore() also apply here."